dask_kubernetes/helm.py

```diff
@@ -253,9 +253,11 @@ def scale(self, n_workers):
         HelmCluster('tcp://localhost:8786', workers=4, threads=24, memory=24.96 GB)
 
         """
-        return self.sync(self._scale, n_workers)
+        return self.sync(self._scale, n_workers, worker_name=worker_name)
 
-    async def _scale(self, n_workers):
+    async def _scale(self, n_workers, worker_name=None):
+        if worker_name:
+            self.worker_name = worker_name
         await self.apps_api.patch_namespaced_deployment(
             name=f"{self.release_name}-{self.chart_name}{self.worker_name}",
```

Comment:
Given that `worker_name` is a transient variable let's not worry about assigning to self. It might make sense to build the deployment name in two steps. Also we shouldn't forget the `-` between the chart name and worker group name.

```suggestion
        deployment = f"{self.release_name}-{self.chart_name}"
        if worker_name:
            deployment += f"-{worker_name}"
        await self.apps_api.patch_namespaced_deployment(
            name=deployment,
```

Reply:
Yeah I would catch `ApiValueError` and then raise something that is useful to the user that called `scale` like `No such group foo`. Maybe a `ValueError` would be an appropriate thing to raise.